cinema/views.py

```diff
@@ -118,18 +140,20 @@ class OrderPagination(PageNumberPagination):
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(generics.ListCreateAPIView):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
 
     def get_queryset(self):
         return Order.objects.filter(user=self.request.user)
 
     def get_serializer_class(self):
-        if self.action == "list":
+        if self.request.method == "GET" and "pk" not in self.kwargs:
```

Comment:
Why? It looks like self.action == "list". Rollback these changes

Reply:
Because self.action doesn't work with generics.ListCreateAPIView.  Action is attribute for object viewsets.ModelViewSet only.

Reply:
I think I can delete this condition because this class doesn't use retrieve.